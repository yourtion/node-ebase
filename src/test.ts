/* tslint:disable:no-conditional-assignment arrow-parens no-console */

import { QueryBuilder } from "squel";
import EBase, { IKVObject } from "./index";

function getAllMethodNames(obj: any) {
  const methods: Set<string> = new Set();
  while ((obj = Reflect.getPrototypeOf(obj))) {
    const keys = Reflect.ownKeys(obj);
    keys.forEach(k => methods.add(k as string));
  }
  return Array.from(methods);
}

const PARAMS_1: IKVObject<any> = {
  _count: [],
  _getByPrimary: ["1", ["a", "c"]],
  _getOneByField: [],
  _deleteByPrimary: ["2"],
  _deleteByField: [{ a: "b" }],
  _insert: [{ a: "b" }],
  _batchInsert: [[{ a: "b", a2: "b2" }]],
  _updateByPrimary: [1, { a: "b2" }],
  _createOrUpdate: [{ c: "1" }],
  _updateByField: [{ a: "b" }, { a: "c" }],
  _incrFields: [1, ["a"], 2],
  _list: [],
  _search: ["Yourtion", ["name"]],
};

class Base<T> extends EBase<T> {
  errorHandler(err: any) { throw err; }
  debugSQL(name: string)  { return (sql: any) => sql; }
  query(sql: string | QueryBuilder, connection: any) { return connection.queryAsync(sql); }
}

describe("Snapshot", () => {
  const base = new Base<string>("test", {}) as IKVObject;
  const methods = getAllMethodNames(base).filter(k => {
    return k.indexOf("_") === 0 && k.indexOf("__") === -1;
  });

  for (const method of methods) {
    test(method, () => {
      const param = PARAMS_1[method];
      const res = base[method](...param);
      expect(res.toString()).toMatchSnapshot();
    });
  }
});
