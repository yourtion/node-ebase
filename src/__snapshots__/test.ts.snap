// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot _batchInsert 1`] = `"INSERT INTO test (a, a2) VALUES ('b', 'b2')"`;

exports[`Snapshot _count 1`] = `"SELECT COUNT(*) AS \\"c\\" FROM test"`;

exports[`Snapshot _createOrUpdate 1`] = `"INSERT INTO test (c) VALUES ('1') ON DUPLICATE KEY UPDATE c = '1'"`;

exports[`Snapshot _deleteByField 1`] = `"DELETE FROM test WHERE (a = 'b' ) LIMIT 1"`;

exports[`Snapshot _deleteByPrimary 1`] = `"DELETE FROM test WHERE (id = '2') LIMIT 1"`;

exports[`Snapshot _getByPrimary 1`] = `"SELECT \`a\`, \`c\` FROM \`test\` WHERE (id = '1') LIMIT 1"`;

exports[`Snapshot _getOneByField 1`] = `"SELECT * FROM \`test\` LIMIT 1"`;

exports[`Snapshot _incrFields 1`] = `"UPDATE test SET a = a + 2 WHERE (id = 1)"`;

exports[`Snapshot _insert 1`] = `"INSERT INTO test (a) VALUES ('b')"`;

exports[`Snapshot _list 1`] = `"SELECT * FROM \`test\` LIMIT 999 OFFSET 0"`;

exports[`Snapshot _search 1`] = `"SELECT * FROM \`test\` WHERE (name like '%Yourtion%') LIMIT 10 OFFSET 0"`;

exports[`Snapshot _updateByField 1`] = `"UPDATE test SET a = 'c' WHERE (a = 'b')"`;

exports[`Snapshot _updateByPrimary 1`] = `"UPDATE test SET a = 'b2' WHERE (id = 1)"`;
